server/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -439,6 +439,27 @@ public ShardRouting initializeShard(
         return initializedShard;
     }
 
+    public ShardRouting relocateOrReinitializeShard(
+        ShardRouting startedShard,
+        String nodeId,
+        long expectedShardSize,
+        RoutingChangesObserver changes
+    ) {
+        if (startedShard.isSearchable()) {
```

Comment:
Possibly also we just do a regular relocate for unsearchable replicas, because that's just like a normal peer recovery....